src/prop/theory_proxy.cpp

```diff
@@ -96,9 +96,38 @@ void TheoryProxy::explainPropagation(SatLiteral l, SatClause& explanation) {
 
   theory::TrustNode tte = d_theoryEngine->getExplanation(lNode);
   Node theoryExplanation = tte.getNode();
-  if (cvc5::options::produceProofs())
+  if (options::produceProofs())
   {
-    d_propEngine->getProofCnfStream()->convertPropagation(tte);
+    // no proof tracking for propagations if we're producing unsat cores
+    if (!options::unsatCoresNew())
```

Comment:
I think this logic should just be put inside of `d_propEngine->getProofCnfStream()->convertPropagation`.  Is there a reason to have this block of code here?

Reply:
I think that wouldn't work because the trust node, when we don't have proofs in the theory engine, does not have the explanation. It just contains the propagated node.

Reply:
Hm, maybe I'd suggest a new method ::convertPropagationTrusted then.

Reply:
I see. Sounds good, will add.